Syntax error on ldi instruction

I have a disassembly that uses the instruction 

```
ldi  a,[hl]
```

rgbasm tells me

```
rgbasm -i src/ -o src/main.o src/main.asm
ERROR:  src/main.asm(1644) :
        syntax error
C:\MinGW\msys\1.0\bin\rgbasm.exe: Assembly aborted in pass 1 (1 errors)!
make: *** [src/main.o] Error 1
```

Right now I am working past that by doing

```
db $2A ; ldi  a,[hl]
```

--- a/src/asm/asmy.y
+++ b/src/asm/asmy.y
@@ -1263,11 +1263,15 @@ z80_ldi			:	T_Z80_LDI T_MODE_HL_IND comma T_MODE_A
 					{ out_AbsByte(0x02|(2<<4)); }
 				|	T_Z80_LDI T_MODE_A comma T_MODE_HL
 					{ out_AbsByte(0x0A|(2<<4)); }
+				|	T_Z80_LDI T_MODE_A comma T_MODE_HL_IND
+					{ out_AbsByte(0x0A|(2<<4)); }
 ;
 
 z80_ldd			:	T_Z80_LDD T_MODE_HL_IND comma T_MODE_A
 					{ out_AbsByte(0x02|(3<<4)); }
 				|	T_Z80_LDD T_MODE_A comma T_MODE_HL
+					{ out_AbsByte(0x0A|(3<<4)); }
+				|	T_Z80_LDD T_MODE_A comma T_MODE_HL_IND
 					{ out_AbsByte(0x0A|(3<<4)); }
 ;
 

